hawkular-bus-common/src/main/java/org/hawkular/bus/common/consumer/RPCBasicMessageListener.java

```diff
@@ -128,12 +131,10 @@ public void onMessage(Message message) {
             msglog.errorFailedToSendResponse(e);
             return;
         } finally {
-            if (consumerConnectionContext != null) {
-                try {
-                    consumerConnectionContext.close();
```

Comment:
Yeah, now that you mention it, I'm not sure why the "consumer" is being closed. The producer (the thing sending the message back to the original sender) should be close. That is what you added here in the finally block. I think this is how it was supposed to look in the first place :)